File: solutions/elixir/lasagna/1/lib/lasagna.ex
defmodule Lasagna do
  def expected_minutes_in_oven(), do: 40

  def remaining_minutes_in_oven(actual), do: expected_minutes_in_oven() - actual

  def preparation_time_in_minutes(layers), do: 2 * layers 

  def total_time_in_minutes(layers, oven_time), do: preparation_time_in_minutes(layers) + oven_time

  def alarm(), do: "Ding!"
end

File: solutions/java/lasagna/2/src/main/java/Lasagna.java
public class Lasagna {

    public int expectedMinutesInOven() {
        return 40;
    }

    public int remainingMinutesInOven(int m) {
        return expectedMinutesInOven() - m;
    }

    public int preparationTimeInMinutes(int l) {
        return l * 2;
    }

    public int totalTimeInMinutes(int l, int m) {
        return preparationTimeInMinutes(l) + m;
    }
}
